manifests/charts/gateway/values.yaml

```diff
@@ -82,6 +82,18 @@ minReadySeconds:
 # Sets the per-pod terminationGracePeriodSeconds setting.
 terminationGracePeriodSeconds:
 
+# Configures the Pod ReadinessProbe
+readinessProbe:
```

Comment:
I recall @howardjohn had a point about why there's no readiness probe on gateway, but I cannot find out the link.

Reply:
The complication is there _is_ a readinessProbe on the [existing gateways chart](https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-ingress/templates/deployment.yaml#L133-L144) - which seems to be what the Istio Operator is using. I found [this note](https://github.com/istio/istio/blob/master/manifests/charts/gateway/README.md#legacy-gateway-helm-charts) in the new `gateway` chart that says we should be avoiding using the `gateways` charts .. but this new `gateway` chart is fundamentally missing settings/features that the older charts seem to have. It's a bit confusing as to why they are different.

Reply:
readiness probe is injected is why. The probe is added when the Pod is created I believe.

Reply:
@howardjohn Ok. So our motivation on the `readinessProbe` is that we have a particular set of ingress gateways where we needed to tune the `readinessProbe.periodSeconds` to prevent an outage. In the old IstioOperator model, we were able to do this with no problem. In the new `gateway` chart mode, there's no mechanism at least in the chart that I see to do that.

After testing, it seems you are right. The `istiod` control plane _does_ inject the `readinessProbe` - only if it does not already exist. If it does exist, it leaves its alone. So it seems to me that the right answer here is to allow explicitly setting the `readinessProbe` via the helm chart ... but default to _no_ readinessProbe to maintain the existing injected behavior.

Is that fair?

Reply:
I've pushed https://github.com/istio/istio/pull/45914/commits/93db64bc162a96f6a57959633714fa5c80e03c36 to remove the default settings entirely, maintaining the default behavior but allowing for an override.